### PR TITLE
Fix slices in Gated Linear Unit (jax.nn.glu) activation

### PR DESCRIPTION
**Pull request by martinsotir**
Add missing division by 2 in the Gated Linear Unit activation function.

### CODE REVIEW

**Comment by martinsotir** [comment]
Also, the axis parameter (default to -1) seems to be ignored in the slicing operation.
I am not sure how to properly implement that (Does splitting along an axis !=1 makes sense? Should splitting along multiple axis be supported ?).

Maybe something along:
```diff
def glu(x, axis=-1):
  """Gated linear unit activation function."""
  size = x.shape[axis]
  assert size % 2 == 0, "axis size must be divisible by 2"

+  slice_a = [slice(None)] * x.ndim
+  slice_a[axis] = slice(None, size // 2)

+  slice_b = [slice(None)] * x.ndim
+  slice_b[axis] = slice(size // 2, None)

-   return x[..., :size // 2] * sigmoid(x[..., size // 2:])
+   return x[tuple(slice_a)] * jax.nn.sigmoid(x[tuple(slice_b)])
```
?

**Comment by jekbradbury** [comment]
Yes, I think that code would work. I think this would too, since jax.numpy is already imported in nn/functions.py:
or
```python
x1, x2 = np.split(x, 2, axis)
return x1 * sigmoid(x2)
```
